lib/ts/axios.ts

```diff
@@ -463,7 +463,26 @@ export default class AuthHttpRequest {
                         );
                         if (response.status === AuthHttpRequestFetch.config.sessionExpiredStatusCode) {
                             logDebugMessage("doRequest: Status code is: " + response.status);
+
+                            /**
+                             * An API may return a 401 error response even with a valid session, causing a session refresh loop in the interceptor.
+                             * To prevent this infinite loop, we break out of the loop after retrying the original request a specified number of times.
+                             * The maximum number of retry attempts is defined by maxRetryAttemptsForSessionRefresh config variable.
+                             */
+                            if (hasExceededMaxSessionRefreshAttempts(config)) {
+                                logDebugMessage(
+                                    `doRequest: Maximum session refresh attempts reached. sessionRefreshAttempts: ${config.__supertokensSessionRefreshAttempts}, maxRetryAttemptsForSessionRefresh: ${AuthHttpRequestFetch.config.maxRetryAttemptsForSessionRefresh}`
+                                );
+                                return response;
```

Comment:
After some discussion I think our interceptors should throw a JS error explaining that this is a refresh loop and any information you think is relevant.

This way it'll be more visible to devs and there should be no way of ignoring this in development.
This means that the site will still break in production if there is a refresh loop, but:
- it used to break anyway
- it doesn't increase the server load as much
- it is much more visible to devs (through error logs) and to users who can report this issue to the devs

Please update the interceptors/tests/changelogs.

Reply:
Done